spec/controllers/miq_report_controller/reports/editor_spec.rb

```diff
@@ -174,6 +174,47 @@
         end
       end
     end
+
+    def non_empty_category
+      cat = FactoryGirl.create(:classification, :name => "non_empty", :description => "Has entries")
+      cat.add_entry(:name => "foo", :description => "Foo")
+      cat.add_entry(:name => "bar", :description => "Bar")
+      cat
+    end
+
+    def empty_category
+      FactoryGirl.create(:classification, :name => "empty", :description => "Zero entries")
+    end
+
+    describe "#categories_hash" do
+      it "returns non-empty categories" do
+        non_empty_category
+        empty_category
+        expect(controller.send(:categories_hash)).to eq("non_empty" => "Has entries")
+      end
+
+      it "omits show: false categories" do
+        non_empty_category.update(:show => false)
```

Comment:
Made the overrides params of the helper passed into the factory, writing once *per example that calls it*.  PTAL.
I suppose I could have `let` that doesn't save and/or shadow the `let` but that sounds complicated, if you still don't like it I'd rather inline.

I don't see a simple way to not write at all (.build / .stub), because the funcs under test query the whole table with scopes etc.